packages/docusaurus-plugin-content-docs/src/sidebars/types.ts

```diff
@@ -183,6 +189,7 @@ export type PropSidebarItemCategory = Expand<
   SidebarItemCategoryBase & {
     items: PropSidebarItem[];
     href?: string;
+    description?: string;
```

Comment:
All those different types is a bit complex to follow, but I thnk this new attribute could go to `SidebarItemCategoryBase` directly, so that it applies to all other category types as well:

We want all these subtypes to have the description attribute:

<img width="358" alt="CleanShot 2022-10-26 at 18 28 26@2x" src="https://user-images.githubusercontent.com/749374/198082214-fc0bf7dc-a339-4590-8ef6-5d77cca5a98a.png">